sqlmesh/core/renderer.py

```diff
@@ -68,7 +70,7 @@ def _render(
         start: t.Optional[TimeLike] = None,
         end: t.Optional[TimeLike] = None,
         execution_time: t.Optional[TimeLike] = None,
-        snapshots: t.Optional[t.Dict[str, Snapshot]] = None,
+        snapshots: t.Optional[t.Iterable[Snapshot]] = None,
```

Comment:
this is a bug, in general, you should probably avoid iterable and use collection, on line 112, the code does "if snapshots", but iterables CANNOT check for empty state. so i'd switch this to collection and all places that pass through to this that you changed from dict to iterable